pkg/controller/replicaset/replica_set.go

```diff
@@ -575,16 +575,6 @@ func (rsc *ReplicaSetController) manageReplicas(filteredPods []*v1.Pod, rs *apps
 					// anything because any creation will fail
 					return nil
 				}
-				if errors.IsTimeout(err) {
-					// Pod is created but its initialization has timed out.
```

Comment:
This comment is wrong but only partially. It assumes that the timeout happens after the pod has been created.

> // If the initialization fails, or if the pod keeps
> // uninitialized for a long time, the informer will not
> // receive any update, and the controller will create a new
> // pod when the expectation expires.

I'm not sure what "the expectation expires" means? That seems to never happen.